### PR TITLE
[GH Issue Summarization] Upgrade to kf v0.4.0-rc.2

### PR DESCRIPTION
**Pull request by texasmichelle**
Use latest ksonnet v0.13.1
Use latest seldon v1alpha2
Remove ksonnet app with full kubeflow platform & replace with components specific to this example.
Remove outdated scripts
Add cluster creation links to Click-to-deploy & kfctl
Add warning not to use the Training with an Estimator guide
Replace commandline with bash for better syntax highlighting
Replace messy port-forwarding commands with svc/ambassador
Add modelUrl param to ui component
Modify teardown instructions to remove the deployment
Fix grammatical mistakes
Combine UI into a single jsonnet file
Remove old version of tensor2tensor component

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/450)

### CODE REVIEW

File: github_issue_summarization/demo/README.md [hunk 1]
```diff
@@ -94,14 +95,14 @@ export ENV=gh-demo-1003
 Set a bucket for the job output
 ```
 DAY=$(date +%Y%m%d)
-ks param set --env=${ENV} tfjob-v1alpha2 output_model_gcs_bucket kubecon-gh-demo
-ks param set --env=${ENV} tfjob-v1alpha2 output_model_gcs_path gh-demo/${DAY}/output
+ks param set --env=${ENV} tfjob output_model_gcs_bucket kubecon-gh-demo
```
Comment (jlewi) [inline github_issue_summarization/demo/README.md]: Did this work?
It looks like the command line arguments changed in a recent refactor of the code
https://github.com/kubeflow/examples/blob/37dd52f49d55ed92675282392caa10abc606b552/github_issue_summarization/notebooks/train.py#L115

The arguments output_model_gcs_bucket no longer exists.

I'm working on a test for TFJob and updating the TFJob spec as needed.

Comment (jlewi) [reply github_issue_summarization/demo/README.md]: Were you using a previously built docker image as opposed to an image built from the current version of the code?

I'd be OK with submitting as is and then fixing in a follow on PR. Once your PR is merged I can sync and make the requisite changes to tfjob.jsonnet to work with the latest version of the code.

Comment (texasmichelle) [reply github_issue_summarization/demo/README.md]: ah - yes, the codelab uses a pretty old image. That seems like a parameter that we want to put back. let's merge this first and then fix.

File: github_issue_summarization/02_distributed_training.md [hunk 2]
```diff
@@ -1,23 +1,26 @@
 # Distributed training using Estimator
 
-Distributed training with keras currently doesn't work; see
+Distributed training with Keras currently does not work. Do not follow this guide
```
Comment (jlewi) [inline github_issue_summarization/02_distributed_training.md]: Perhaps in a follow on PR we should delete the code related to using TF.estimator? 
No one seems to be interested in fixing it. We can rely on git history to provide a version of the old code if anyone is interested in picking it up.

File: github_issue_summarization/02_training_the_model_tfjob.md [hunk 3]
```diff
@@ -38,24 +41,24 @@ Submit the training job
...
 
-If you are running on GCS you can train using GCS to store the input
+If you are using GCS, you can train using GCS to store the input
 and the resulting model.
 
-### GCS Service account
+### GCS service account
 
-* Create a service account which will be used to read and write data from the GCS Bucket.
+* Create a service account that will be used to read and write data from the GCS bucket.
```
Comment (jlewi) [inline github_issue_summarization/02_training_the_model_tfjob.md]: If they used kfctl.sh or click to deploy there should already be a service account in the namespace "kubeflow" that they could use.

However if they wanted to use a different namespace they would need to perform these steps.

File: github_issue_summarization/02_training_the_model_tfjob.md [hunk 4]
```diff
@@ -74,12 +77,12 @@ kubectl --namespace=${NAMESPACE} create secret generic user-gcp-sa --from-file=u
...
+[ks_app](ks_app) contains a ksonnet app to deploy the TFJob.
 
-Set the appropriate params for the tfjob component
+Set the appropriate params for the tfjob component:
 
-```commandline
-cd ks-kubeflow
+```bash
+cd ks_app
 ks param set tfjob namespace ${NAMESPACE} --env=${KF_ENV}
```
Comment (jlewi) [inline github_issue_summarization/02_training_the_model_tfjob.md]: I think namespace is set based on the environment and is no longer an explicit parameter.
https://github.com/kubeflow/examples/blob/79904082079b56a7c0fd8a009855d8fa6be97bb9/github_issue_summarization/ks-kubeflow/components/tfjob-v1alpha2.jsonnet#L7

So we should probably delete the line 
ks param set tfjob namespace ...

If users want to set the namespace they should do this by setting the namespace on their environment (they may need to create a new environment for that).

File: github_issue_summarization/demo/README.md (hunk 1 above)
Comment (jlewi) [reply github_issue_summarization/demo/README.md]: The new version of the code supports gcs you just use a full GCS path for the input/output arguments.
The code uses the prefix "gs://" to detect that its a a GCS path and not a local filesystem path.

**Comment by jlewi** [comment]
Left a couple comments. I'd be ok with submitting this as is though and doing additional fixes in a follow on PR.
Renaming the app from kubeflow to ks_app is a big change so the sooner we get it in the better (e.g. I can rebase my pending PRs adding tests on it). So I'll lgtm this and if that sounds good to you, just remove the hold

/lgtm
/approve
/hold

I think a bigger issue is what to do about the tf.estimator and distributed training code. I think having non-working code isn't really helpful.

I'm not seeing much movement on getting distributed training to work with this example:
kubeflow/examples#196

We might have better luck investing in another example to provide a good distributed training example, rather than trying to make distributed training work with this particular model.

In particular we have the object detection example which supports distributed training
https://github.com/kubeflow/examples/tree/master/object_detection

**Comment by texasmichelle** [comment]
Thanks for the review - I'll address the outstanding issues after the holidays & finish updating the codelabs/qwiklabs to v0.4.0.

/hold cancel

**Comment by Vanithanow** [comment]
[GH Issue Summarization] Upgrade to kf v0.4.0-rc.2 (#450)

**Pull request merged**